frontend/src/features/admin/agencies/EditAgencyPage.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from 'react';
+import { Navbar, Container, Row, ButtonToolbar, Button } from 'react-bootstrap';
+import { Form, Input, Select, SelectOption } from '../../../components/common/form';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+import { IAgencyDetail } from 'interfaces';
+import { Formik, Field } from 'formik';
+import * as API from 'constants/API';
+import './EditAgencyPage.scss';
+import { Label } from 'components/common/Label';
+import { useHistory } from 'react-router-dom';
+import TooltipIcon from 'components/common/TooltipIcon';
+import useCodeLookups from 'hooks/useLookupCodes';
+import { ILookupCode } from 'actions/lookupActions';
+import {
+  createAgency,
+  deleteAgency,
+  fetchAgencyDetail,
+  getUpdateAgencyAction,
+} from 'actionCreators/agencyActionCreator';
+import { FaArrowAltCircleLeft } from 'react-icons/fa';
+import { clear } from 'actions/genericActions';
+import * as actionTypes from 'constants/actionTypes';
+import GenericModal from 'components/common/GenericModal';
+
+interface IEditAgencyPageProps {
+  id: number;
+  match?: any;
+}
+
+const EditAgencyPage = (props: IEditAgencyPageProps) => {
```

Comment:
Would love to see more jsdoc in our frontend.  We don't document things well here.

Reply:
Will add some more!